Sweetener.Logging/Templates/TemplateBuilder.cs

```diff
@@ -0,0 +1,251 @@
+                    break;
+                }
+                else
+                {
+                    // Unexpected character!
+                    name = null;
+                    break;
+                }
+
+                // Move onto the next character
+                reader.Read();
+            }
+
+            // Did we reach the end without breaking or was there an unexpected character?
+            if (name == null)
+                throw new FormatException("Input string was not in a correct format");
+
+            return TemplateParameterName.Parse(name);
+        }
+
+        private int? ReadAlignment(TextReader reader)
+        {
+            // Read the alignment between the comma ',' and either the colon ':' or closed curly brace '}'
+            if (reader.Peek() == ',')
+            {
+                StringBuilder alignmentBuilder = new StringBuilder();
+
+                int next;
+                reader.Read(); // Move past the comma
+                while ((next = reader.Peek()) != -1)
+                {
+                    // There are no recognized escape sequences in the alignment
+                    if (next == ':' || next == '}')
+                        break;
+
+                    alignmentBuilder.Append((char)reader.Read());
+                }
+
+                return int.Parse(alignmentBuilder.ToString());
+            }
+
+            return null;
+        }
+
+        private string ReadFormat(TextReader reader)
+        {
+            // Read the format between the comma ':' and the closed curly brace '}'
+            if (reader.Peek() == ':')
+            {
+                StringBuilder formatBuilder = new StringBuilder();
+
+                int next;
+                reader.Read(); // Move past the colon
+                while ((next = reader.Peek()) != -1)
+                {
+                    // '}' cannot be escaped, so there is no need to special-case escape sequences
+                    if (next == '}')
+                        break;
+
+                    formatBuilder.Append((char)reader.Read());
+                }
+
+                return formatBuilder.ToString();
+            }
+
+            return null;
+        }
+
+        private readonly struct LogEntryTemplate<T> : ILogEntryTemplate<T>
+        {
+            private readonly string _format;
+            private readonly string _template;
+
+            private static readonly Process s_currentProcess = Process.GetCurrentProcess();
+
+            public LogEntryTemplate(string template, string format)
+            {
+                // TODO: While the originalTemplate is really bloat in production, it is
+                //       very helpful to have when debugging and for unit testing.
+                //       That said, is there a better mechanism that doesn't involve
+                //       different code between debug/release configuration?
+                _format   = format;
+                _template = template;
```

Comment:
Like the comment says, I use this field exclusively for unit testing purposes currently. Given there aren't many loggers allocated for a single application, it doesn't seem too harmful.